config/bootstrap.php

```diff
@@ -64,6 +64,20 @@
 use Cake\Utility\Inflector;
 use Cake\Utility\Security;
 
+/**
+ * Read .env file if APP_NAME is not set.
+ *
+ * You can remove this block if you do not want to use environment
+ * variables for configuration when deploying.
+ */
+if (!env('APP_NAME') && file_exists(CONFIG . '.env')) {
+    $dotenv = new \josegonzalez\Dotenv\Loader([CONFIG . '.env']);
```

Comment:
Agreed however many apps could probably do without mobiledetect whereas for env we could see if it becomes the new default including docs in the book. 

Well but once that happens it can still be moved it inside the cakephp repo.